conans/client/conan_api.py

```diff
@@ -457,6 +462,40 @@ def download(self, reference, remote_name=None, package=None, recipe=False):
         else:
             raise ConanException("Provide a valid full reference without wildcards.")
 
+    @api_method
+    def workspace_install(self, path, settings=None, options=None, env=None,
+                          remote_name=None, build=None, profile_name=None,
+                          update=False, cwd=None):
+        cwd = cwd or get_cwd()
+        abs_path = os.path.normpath(os.path.join(cwd, path))
+
+        workspace = Workspace(abs_path, self._cache)
+        graph_info = get_graph_info(profile_name, settings, options, env, cwd, None,
+                                    self._cache, self._user_io.out)
+
+        self._user_io.out.info("Configuration:")
+        self._user_io.out.writeln(graph_info.profile.dumps())
+
+        self._cache.editable_packages.override(workspace.get_editable_dict())
+
+        recorder = ActionRecorder()
+        deps_graph, _ = self._graph_manager.load_graph(workspace.root, None, graph_info, build,
+                                                       False, update, remote_name, recorder)
+
+        print_graph(deps_graph, self._user_io.out)
+
+        # Inject the generators before installing
+        for node in deps_graph.nodes:
+            if node.recipe == RECIPE_EDITABLE:
+                generators = workspace[node.ref].generators
+                if generators is not None:
+                    node.conanfile.generators = generators
```

Comment:
Do we need to override? Can we append to existing ones? There could be recipes using custom generators (like Qt does) to create needed files.